src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaDeclBase.cs

```diff
@@ -19,20 +20,20 @@ internal enum Use
         };
 
         protected XmlQualifiedName name = XmlQualifiedName.Empty;
-        protected string prefix;
+        protected string? prefix;
         protected bool isDeclaredInExternal = false;
         protected Use presence;     // the presence, such as fixed, implied, etc
 
-        protected XmlSchemaType schemaType;
-        protected XmlSchemaDatatype datatype;
+        protected XmlSchemaType? schemaType;
+        protected XmlSchemaDatatype datatype = null!;
 
-        protected string defaultValueRaw;       // default value in its original form
-        protected object defaultValueTyped;
+        protected string? defaultValueRaw;       // default value in its original form
+        protected object? defaultValueTyped;
 
         protected long maxLength; // dt:maxLength
         protected long minLength; // dt:minLength
 
-        protected List<string> values;    // array of values for enumerated and notation types
+        protected List<string> values = null!;    // array of values for enumerated and notation types
```

Comment:
I'm assuming you're forcing this because all inheriting types are initializing the field? If that's the case, would it maybe make sense to slightly refactor so that the constructor accepts it as a parameter?

Or alternatively leave as nullable and assert non-nullability on the two Check methods that reference it?

Reply:
Generally this is an abstract type which derived types should fill this field as non-null since all call-sites expect it. Nullable feature  doesn't have a way to express it other than suppressing the warning. Since this is internal type I don't think it's worth adding `!` or Debug.Assert all callsites especially given how large this code base is.

Reply:
Ok, this logic is super convoluted but this is actually nullable